src/ol/control/ZoomSlider.js

```diff
@@ -366,8 +368,8 @@ class ZoomSlider extends Control {
     if (!mapEvent.frameState) {
       return;
     }
-    if (!this.sliderInitialized_) {
-      this.initSlider_();
+    if (!this.sliderInitialized_ && !this.initSlider_()) {
+      return;
```

Comment:
The slider can still get out of sync with the map, e.g. when a view with different resolution settings is assigned to the map.